bin/mympirun.py

```diff
@@ -40,83 +40,139 @@
 
     found_mpi_names = [x.__name__ for x in found_mpi]
     found_sched_names = [x.__name__ for x in found_sched]
 
     if mo.options.showmpi:
         fancylogger.setLogLevelInfo()
-        _logger.info("Found MPI classes %s" % (", ".join(found_mpi_names)))
-        raise ExitException("Exit from showmpi")
+        _logger.info("mympirun.py - Found MPI classes %s" %
+            (", ".join(found_mpi_names)))
+        sys.exit(0)
 
     if mo.options.showsched:
         fancylogger.setLogLevelInfo()
-        _logger.info("Found Sched classes %s" % (", ".join(found_sched_names)))
-        raise ExitException("Exit from showsched")
+        _logger.info("mympirun.py - Found Sched classes %s" %
+            (", ".join(found_sched_names)))
+        sys.exit(0)
 
     if mpi is None:
-        mo.parser.print_shorthelp()
-        mo.log.raiseException(("No MPI class found (scriptname %s; ismpirun %s). Please use mympirun through one "
-                               "of the direct calls or make sure the mpirun command can be found. "
-                               "Found MPI %s") % (scriptname, ismpirun, ", ".join(found_mpi_names)))
+        mo.log.raiseException(
+            ("No MPI class found that supports scriptname %s; ismpirun %s). "
+             "Please use mympirun through one of the direct calls or make sure "
+             "the mpirun command can be found. Found MPI %s") %
+            (scriptname, ismpirun, ", ".join(found_mpi_names)))
     else:
-        mo.log.debug("Found MPI class %s (scriptname %s; ismpirun %s)" % (mpi.__name__, scriptname, ismpirun))
+        mo.log.debug("Found MPI class %s (scriptname %s; ismpirun %s)" %
+                     (mpi.__name__, scriptname, ismpirun))
 
     if sched is None:
-        mo.log.raiseException("No sched class found (options.schedtype %s ; found Sched classes %s)" %
-                              (mo.options.schedtype, ", ".join(found_sched_names)))
+        mo.log.raiseException(
+            ("No sched class found (options.setsched %s ; found Sched "
+             "classes %s)") % (mo.options.setsched,
+             ", ".join(found_sched_names)))
     else:
-        mo.log.debug("Found sched class %s from options.schedtype %s (all Sched found %s)" %
-                     (sched.__name__, mo.options.schedtype, ", ".join(found_sched_names)))
+        mo.log.debug(
+            ("Found sched class %s from options.setsched %s (all Sched "
+             "found %s)") % (sched.__name__, mo.options.setsched,
+             ", ".join(found_sched_names)))
 
     return mpi, sched, mo
 
 
+def import_implementations(module):
+    """searches and imports python files in the some folder as a module
+
+    Arguments:
+        module      --  the module
+    """
+
+    _logger.debug("mympirun.py - import_implementations()")
+
+    # get the paths of all the python files in the module folder
+    modulepaths = glob.glob(os.path.dirname(module.__file__) + "/*.py")
```

Comment:
I don't really like this. Can you have a look at http://stackoverflow.com/a/16853487 for example? Not sure if that will work but it avoids references to `__init__` and such.